InsiteTeamTask/Repositories/DataRepository.cs

```diff
@@ -17,47 +15,136 @@ public List<Attendance> GetAttendanceListFor(int gameNumber)
             List<Ticket> tickets = service.Tickets().ToList();
             List<Product> products = service.Products().ToList();
 
+            List<Attendance> attendanceList = new List<Attendance>();            
+
+            var filteredMembers = members.Where(member => {
+                    var filteredProduct = products.Single(product => product.Id == member.ProductId);
+                
+                    return filteredProduct.GameId == gameNumber;
+                })
+                .Select(member =>
+                {
+                    return new Attendance
+                    {
+                        Barcode = "N/A",
+                        MemberId = member.Id
+                    };
+                });
+
+            attendanceList.AddRange(filteredMembers);
+
+            var filteredTickets = tickets.Where(ticket => {
+                    var filteredProduct = products.Single(product => product.Id == ticket.ProductId);
+
+                    return filteredProduct.GameId == gameNumber;
+                })
+                .Select(ticket =>
+                {
+                    return new Attendance
+                    {
+                        Barcode = ticket.Barcode,
+                        MemberId = 0
+                    };  
+                });
+
+            attendanceList.AddRange(filteredTickets);
+
+            return attendanceList;
```

Comment:
We only asked for methods to get Attendance by:
- Game and Season IDs
- Product Code

so I'm going to skip reviewing this method